Bottom: efeca456a3826ce2a412d7ac84c7b409f81d57ab
Top:    f47b9003821d5799a9a51c8ae5ab8b6acc2ee913
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-08-31 14:47:35 -0700

Refresh of addrtype

---

diff --git a/examples/cmatose.c b/examples/cmatose.c
index c00d28f..167f509 100644
--- a/examples/cmatose.c
+++ b/examples/cmatose.c
@@ -641,7 +641,7 @@ int main(int argc, char **argv)
 	int op, ret;
 
 	hints.ai_port_space = RDMA_PS_TCP;
-	while ((op = getopt(argc, argv, "s:b:f:c:C:S:t:p:m")) != -1) {
+	while ((op = getopt(argc, argv, "s:b:f:P:c:C:S:t:p:m")) != -1) {
 		switch (op) {
 		case 's':
 			dst_addr = optarg;
@@ -655,11 +655,17 @@ int main(int argc, char **argv)
 			} else if (!strncasecmp("gid", optarg, 3)) {
 				hints.ai_flags = RAI_NUMERICHOST | RAI_FAMILY;
 				hints.ai_family = AF_IB;
-				hints.ai_port_space = RDMA_PS_IB;
 			} else if (strncasecmp("name", optarg, 4)) {
 				fprintf(stderr, "Warning: unknown address format\n");
 			}
 			break;
+		case 'P':
+			if (!strncasecmp("ib", optarg, 2)) {
+				hints.ai_port_space = RDMA_PS_IB;
+			} else if (strncasecmp("tcp", optarg, 3)) {
+				fprintf(stderr, "Warning: unknown port space format\n");
+			}
+			break;
 		case 'c':
 			connections = atoi(optarg);
 			break;
@@ -685,6 +691,8 @@ int main(int argc, char **argv)
 			printf("\t[-b bind_address]\n");
 			printf("\t[-f address_format]\n");
 			printf("\t    name, ip, ipv6, or gid\n");
+			printf("\t[-P port_space]\n");
+			printf("\t    tcp or ib\n");
 			printf("\t[-c connections]\n");
 			printf("\t[-C message_count]\n");
 			printf("\t[-S message_size]\n");
diff --git a/man/ucmatose.1 b/man/ucmatose.1
index baf5837..95dc2d6 100644
--- a/man/ucmatose.1
+++ b/man/ucmatose.1
@@ -4,9 +4,11 @@ ucmatose \- RDMA CM connection and simple ping-pong test.
 .SH SYNOPSIS
 .sp
 .nf
-\fIucmatose\fR [-s server_address] [-b bind_address] [-f address_format]
+\fIucmatose\fR [-s server_address] [-b bind_address]
+		[-f address_format] [-P port_space]
 		[-c connections] [-C message_count] [-S message_size]
-\fIucmatose\fR -s server_address [-b bind_address] [-f address_format]
+\fIucmatose\fR -s server_address [-b bind_address]
+		[-f address_format] [-P port_space]
 		[-c connections] [-C message_count] [-S message_size] [-t tos]
 .fi
 .SH "DESCRIPTION"
@@ -29,6 +31,11 @@ address, or an IPv6 address.  This option may be used to indicate that
 a specific address format has been provided.  Supported address_format
 values are: name, ip, ipv6, and gid.
 .TP
+\-P port_space
+Specifies the port space for the connection.  Be default, the port space
+is the RDMA TCP port space. (Note that the RDMA port space may be separate
+from that used for IP.)  Supported port_space values are: tcp and ib.
+.TP
 \-c connections
 The number of connections to establish between the client and server.
 (default 1)
